crates/autopilot/src/driver_model.rs

```diff
@@ -150,6 +154,30 @@ pub mod solve {
         pub submission_address: H160,
     }
 
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub enum FeePolicy {
+        /// Applies to limit orders only.
+        /// This fee should be taken if the solver provided good enough solution
+        /// that even after the surplus fee is taken, there is still more
+        /// surplus left above whatever the user expects [order limit price
+        /// or best quote, whichever is better for the user].
```

Comment:
The fact that this applies to limit orders only is not guaranteed (and a detail of the first policy that is being tested). We can and likely will charge positive quote deviation also for market orders.
I also wouldn't mention the surplus fee at all here since it's confusing. If the user received more than expected, solvers need to pay the protocol.

```suggestion
        /// If the order receives more than expected (positive deviation from quoted amounts)
        /// pay the protocol a factor of the achieved improvement.
```

Reply:
added